server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalTopHits.java

```diff
@@ -184,9 +185,42 @@ protected boolean mustReduceOnSingleInternalAgg() {
     public Object getProperty(List<String> path) {
         if (path.isEmpty()) {
             return this;
+        }
+        if (path.size() != 1) {
+            throw new IllegalArgumentException(
+                "property paths for top_hits ["
+                    + getName()
+                    + "] can only contain a single field in _source, score or sort values, got "
+                    + path
+            );
+        }
+
+        // Supported property prefixes.
+        final String SOURCE = "_source";
+        final String SORT_VALUE = "_sort";
+        final String SCORE = "_score";
+
+        String[] tokens = path.get(0).toLowerCase(Locale.ROOT).split(":|>|\\.");
+        assert searchHits.getHits().length == 1 : "property paths should only resolve against top_hits with size == 1.";
```

Comment:
Maybe we should throw an `IllegalArgumentException` here instead of the assert?
I think this is the only place where we can check whether an pipeline agg access a top_hits agg that has multiple hits (meaning size is not equal to 1).